### PR TITLE
feat: add join time to CL position creation response

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #4627 

## What is the purpose of the change

Adds a join_time response to creating a concentrated liquidity position. Without this in the response, contracts such as Mars Rover will not be able to retrieve an end user's position via the key.

## Brief Changelog

- Added a new field called joinTime to the create position response. This field represents the timestamp when the CL position was created.
- Updated the unit tests to ensure the joinTime is returned in all respective responses after CL position creation

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/lp_test.go [hunk 1]
```diff
@@ -236,7 +236,7 @@ func (s *KeeperTestSuite) TestCreatePosition() {
 			poolBalancePrePositionCreation := s.App.BankKeeper.GetAllBalances(s.Ctx, pool.GetAddress())
 
 			// System under test.
-			asset0, asset1, liquidityCreated, err := clKeeper.CreatePosition(s.Ctx, tc.poolId, s.TestAccs[0], tc.amount0Desired, tc.amount1Desired, tc.amount0Minimum, tc.amount1Minimum, tc.lowerTick, tc.upperTick, tc.freezeDuration)
+			asset0, asset1, liquidityCreated, _, err := clKeeper.CreatePosition(s.Ctx, tc.poolId, s.TestAccs[0], tc.amount0Desired, tc.amount1Desired, tc.amount0Minimum, tc.amount1Minimum, tc.lowerTick, tc.upperTick, tc.freezeDuration)
```
Comment (p0mvn) [inline x/concentrated-liquidity/lp_test.go:239]: We should have a an assertion verifying that the `joinTime` equals to block time

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM

**Review by p0mvn — APPROVED** [review]
LGTM after the requested assertion is added.

Do we have docs for migrations logic in `architectture.md`? If not, could you please cut an issue for that

File: x/concentrated-liquidity/lp_test.go (hunk 1 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/lp_test.go]: added! https://github.com/osmosis-labs/osmosis/pull/4633/commits/0fe55b6e4c1a16e8464e58dfd27d1b27b4f0212c

**Pull request merged**